ERCS/erc-XXXX.md

```diff
@@ -77,134 +81,166 @@ Example TWIT manifest at <https://my-crypto-dapp.org/twit.json>:
         }
+      }
     }
+  }
 }
 ```
 
-#### Example implementation
+#### RPC method
 
-We propose to rely on algorithms supported by [SubtleCrypto](https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto) as they are present in every browsers.
+The parameters of `wallet_signedRequest` are specified by this TypeScript interface:
 
-We recommend adding new method `secureRequest` to EIP-1193 as it would provide a simple, yet extensive, implementation for secure calls. In addition, it provides
-dApp maintainers/devloppers with an explicit way to ask wallets to verify the integrity of the payload, as opposed to an implicit implementations relying on bit
-packing of the arguments.
+```typescript
+type RequestPayload<Params> = { method: string; params: Params };
 
-Example:
+type SignedRequestParameters<Params> = [
+  requestPayload: RequestPayload<Params>,
+  signature: `0x${string}`, // TODO: Is this the format we want?
+  keyId: string,
+];
+```
+
+Here's a non-normative example of calling `wallet_signedRequest` using the EIP-1193 provider interface:
 
 ```typescript
-interface RequestArguments { // this interface comes from EIP-1193
-  readonly method: string;
-  readonly params?: readonly unknown[] | object;
-}
-async function secureRequest(requestArg: RequestArguments, signature: byte[], keyId:string): Promise<unknown>;{
-  // 0. get the domain of the sender.tab.url
-  const domain = getActiveTabDomain()
-
-  // 1. get the manifest for the current domain
-  // One can use RFC 8484 for the actual DNS-over-HTTPS specification (see https://datatracker.ietf.org/doc/html/rfc8484).
-  // here we are doing it with DoHjs
-  //
-  // Note: this step is optional, and one could opt to go directly to the well-known address first at `domain + '/.well-known/twit.json'`
+const keyId = '1';
+const requestPayload: RequestPayload<TransactionParams> = {
+  method: 'eth_sendTransaction',
+  params: [
+    {
+      /* ... */
+    },
+  ],
+};
+const signature: `0x${string}` = await getSignature(requestPayload, keyId);
+
+// Using the EIP-1193 provider interface
+const result = await ethereum.request({
+  method: 'wallet_signedRequest',
+  params: [requestPayload, signature, keyId],
+});
+```
+
+#### Signature verification
+
+1. Upon receiving an EIP-1193 call, the wallet MUST check of the existence of the TWIT manifest for the `sender.tab.url` domain
+   a. The wallet MUST verify that the manifest is hosted on the `sender.tab.url` domain
+   b. The wallet SHOULD find the DNS TXT record to find the manifest location
+   b. The wallet MAY first try the `/.well-known/twit.json` location
+2. If TWIT is NOT configured for the `sender.tab.url` domain, then proceed as usual
+3. If TWIT is configured and the `request` method is used, then the wallet SHOULD display a visible and actionable warning to the user
+   a. If the user opts to ignore the warning, then proceed as usual
+   b. If the user opts to cancel, then the wallet MUST cancel the call
```

Comment:
I downgraded the previous "MUST" in this section to a "SHOULD", because we can't actually force wallets to do this. I also have to pay it some more thought in order to make up my mind about the suggestion in general.

I'm also noticing that this section needs some edits to align with some of my other changes, but that can be done in a follow-up.